graphrag/index/verbs/genid.py

```diff
@@ -52,6 +52,19 @@ def genid(
     """
     data = cast(pd.DataFrame, input.source.table)
 
+    output = genid_df(data, to, method, hash)
+
+    return TableContainer(table=output)
+
+
+def genid_df(
+    input: pd.DataFrame,
+    to: str,
+    method: str = "md5_hash",
+    hash: list[str] = [],  # noqa A002
```

Comment:
I'd rewrite it this way:

```python
def genid_df(
    input: pd.DataFrame,
    to: str,
    method: str = "md5_hash",
    hash: list[str] | None = None,
):
    data = input
    
    match method:
        case "md5_hash":
            if not hash:
                msg = 'Must specify the "hash" columns to use md5_hash method'
                raise ValueError(msg)
            data[to] = data.apply(lambda row: gen_md5_hash(row, hash), axis=1)
        case "increment":
            data[to] = data.index + 1
        case _:
            msg = f"Unknown method {method}"
            raise ValueError(msg)
    
    return data

```

I changed the `if len(hash) == 0:` to `if not hash:` (this will check if hash is None and if len(hash) > 0) 

That way if the hash param is not provided and the method is md5_hash it will raise the ValueError.

 Also using match/case instead of if/elif/else...